### ISSUE TITLE
Precondition failure from alire-publish.adb

### ISSUE THREAD

**Issue by pmderodat** [alire-project/alire#606]
Hello,

While trying to publish a crate for a example program using Libadalang, I got a crash:

```
$ alr publish git+https://github.com/AdaCore/libadalang 04b7560f2302839b4c172197826ea652a5ad2d0a -d

Publishing assistant: step 1 of 6: Verify origin URL
[…]
✓ Build succeeded.

Publishing assistant: step 5 of 6: User review
stderr: SYSTEM.ASSERTIONS.ASSERT_FAILURE
stderr: failed precondition from alire-publish.adb:456
stderr: raised SYSTEM.ASSERTIONS.ASSERT_FAILURE : failed precondition from alire-publish.adb:456
[/home/pmderodat/usr/bin/alr]
System.Assertions.Raise_Assert_Failure at s-assert.adb:46
Alire.Publish.Show_And_Confirm at alire-publish.adb:456
Alire.Publish.Start_At at alire-publish.adb:722
Alire.Publish.Remote_Origin at alire-publish.adb:880
Alr.Commands.Publish.Execute at alr-commands-publish.adb:76
Alr.Commands.Execute_By_Name at alr-commands.adb:743
Alr.Commands.Execute at alr-commands.adb:713
Alr.Main at alr-main.adb:28
Main at b__alr-main.adb:1851
[/lib/x86_64-linux-gnu/libc.so.6]
0x7f42244bd1e1
[/home/pmderodat/usr/bin/alr]
0x430538 _start at ???
0xfffffffffffffffe

error: failed precondition from alire-publish.adb:456
error: alr encountered an unexpected error, re-run with -d for details.
```
I’m using:
```
$ alr version
Alr version: 0.7.0-dev
Alire Library version: 0.7.0-dev
alr status is (valid) (0 releases indexed) (loaded in 0.026s)
config folder is /home/pmderodat/.config/alire
source folder is /home/pmderodat/.config/alire/alire
interaction flags are: force:FALSE not-interactive:FALSE
alr root is lal_highlight=21.0.0
alr root detection has settled on path: /home/pmderodat/_git/libadalang
alr is finding 0 GPR project files
alr session state is [VALID]
alr compiled on [2020-10-21 12:41:07] with GNAT version [Community 2020 (20200818-93)]
platform fingerprint: Linux Bits_64 Ubuntu
platform properties: UBUNTU LINUX NATIVE SYSTEM BITS_64
community index required branch: stable-0.4
```

There commands were executed at the root a clone of the repository above, with the commit passed to `alr` checked out, and here’s my `alire.toml`:
```
name = "lal_highlight"
description = "Shiny new project"
version = "21.0.0"

authors = ["AdaCore"]
maintainers = ["Pierre-Marie de Rodat <pmderodat@kawie.fr>"]
maintainers-logins = ["pmderodat"]

project-files = ["contrib/highlight/highlight.gpr"]
executables = ["highlight", "ada2web"]

[[depends-on]]  # This line was added by `alr with`
libadalang = "^21.0"  # This line was added by `alr with`
```

Please let me know if you need help to reproduce. Thanks!

**Comment by Fabien-Chouteau** [comment]
> Alr version: 0.7.0-dev
> Alire Library version: 0.7.0-dev

So this is an `alr` built from sources?

**Comment by pmderodat** [comment]
No, it was installed from a GitHub binary release.

**Comment by mosteo** [comment]
> So this is an `alr` built from sources?

This is then my fault at some point when I retagged the `0.7` release. Incidentally, I was going to propose to release `0.7.1` soon once our pipeline of recent bug reports is empty, so we can have this [improper version] fixed too.

**Comment by mosteo** [comment]
> I got a crash:

This is likely [fixed by] #595 (originally reported in #569, #593). If you have the time to test with a build from `master` we will know for sure; otherwise I'll try to reproduce later.

**Comment by mosteo** [comment]
I just checked and this is indeed duplicate of #569 and fixed in `master`, so closing. Thanks for the report!
